src/main/java/com/google/devtools/build/lib/skyframe/toolchains/SingleToolchainResolutionFunction.java

```diff
@@ -181,101 +193,120 @@ private static SingleToolchainResolutionValue resolveConstraints(
       }
       if (!nonmatchingSettings.isEmpty()) {
         debugMessage(
-            eventHandler,
-            "    Type %s: %s platform %s: Rejected toolchain %s; mismatching config settings: %s",
-            toolchainType.toolchainType(),
-            "target",
-            targetPlatform.label(),
+            resolutionTrace,
+            "  Rejected toolchain %s; mismatching config settings: %s",
```

Comment:
I like the core idea here: accumulate everything at once, use newlines and indents to format it clearly. Instead of having the indents implicit in the template message, can you add an `indent` parameter to `debugMessage`, and use that instead of putting spaces in the actual template?

Reply:
I can certainly do that.